tests/mlflow/sagemaker/test_model_export.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import print_function
+
+import os
+import pickle
+import requests
+from subprocess import Popen, PIPE, STDOUT
+import tempfile
+import time
+import unittest
+
+import sklearn.datasets as datasets
+import sklearn.linear_model as glm
+
+import numpy as np
+import pandas as pd
+
+from mlflow.utils.file_utils import TempDir
+from mlflow import pyfunc
+
+import mlflow.sagemaker
+
+
+def load_pyfunc(path):
+    with open(path, "rb") as f:
+        return pickle.load(f)
+
+
+CONDA_ENV = """
+name: mlflow-env
+channels:
+  - anaconda
+  - defaults
+dependencies:
+  - python={python_version}
+
+"""
+
+
+class TestModelExport(unittest.TestCase):
+    def setUp(self):
+        self._tmp = tempfile.mkdtemp()
+        iris = datasets.load_iris()
+        self._X = iris.data[:, :2]  # we only take the first two features.
+        self._y = iris.target
+        self._iris_df = pd.DataFrame(self._X, columns=iris.feature_names[:2])
+        self._linear_lr = glm.LogisticRegression()
+        self._linear_lr.fit(self._X, self._y)
+        self._linear_lr_predict = self._linear_lr.predict(self._X)
+        os.environ["LC_ALL"] = "en_US.UTF-8"
+        os.environ["LANG"] = "en_US.UTF-8"
+        mlflow_root = os.environ.get("MLFLOW_HOME") if "MLFLOW_HOME" in os.environ \
+            else os.path.dirname(os.path.dirname(os.path.abspath(mlflow.__file__)))
+        # "/home/travis/build/databricks/mlflow"
+        print("Building mlflow Docker image with MLFLOW_HOME =", mlflow_root)
+        mlflow.sagemaker.build_image(mlflow_home=mlflow_root)
+
+    def test_model_export(self):
+        path_to_remove = None
+        try:
+            with TempDir(chdr=True, remove_on_exit=False) as tmp:
+                path_to_remove = tmp._path
+                # NOTE: Changed dir to temp dir and use relative paths to get around the way temp
+                # dirs are handled in python.
+                model_pkl = tmp.path("model.pkl")
+                with open(model_pkl, "wb") as f:
+                    pickle.dump(self._linear_lr, f)
+                input_path = tmp.path("input_model")
+                conda_env = "conda.env"
+                import sys
+                python_version = sys.version.split("|")[0].strip()
```

Comment:
It's better to use sys.version to pull out the components; for example Mac OS's Python doesn't have any `|` in it